pkg/agent/agent.go

```diff
@@ -496,27 +487,77 @@ func (c *agentController) CreateAddOnPlacementScore() {
 			Name:      util.HostedClusterScoresResourceName,
 			Namespace: c.clusterName,
 		},
-		Status: clusterv1alpha1.AddOnPlacementScoreStatus{
-			Scores: items,
-		},
 	}
 
-	_, err = controllerutil.CreateOrUpdate(context.TODO(), c.hubClient, addOnPlacementScore, func() error { return nil })
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), c.hubClient, addOnPlacementScore, func() error { return nil })
 	if err != nil {
 		// just log the error. it should not stop the rest of reconcile
 		c.log.Error(err, fmt.Sprintf("failed to create or update the addOnPlacementScore resource in %s", c.clusterName))
-		return
+		return err
 	}
 
-	addOnPlacementScore.Status.Scores = items
-	err = c.hubClient.Status().Update(context.TODO(), addOnPlacementScore, &client.UpdateOptions{})
+	listopts := &client.ListOptions{}
+	hcList := &hyperv1alpha1.HostedClusterList{}
+	err = c.spokeUncachedClient.List(context.TODO(), hcList, listopts)
 	if err != nil {
 		// just log the error. it should not stop the rest of reconcile
-		c.log.Error(err, fmt.Sprintf("failed to update the addOnPlacementScore status in %s", c.clusterName))
-		return
+		c.log.Error(err, "failed to get HostedCluster list")
+
+		conditions := []metav1.Condition{
+			{
+				LastTransitionTime: metav1.Time{Time: time.Now()},
+				Type:               "HostedClusterCountUpdated",
+				Status:             metav1.ConditionFalse,
+				Reason:             "HostedClusterCountFailed",
+				Message:            err.Error(),
+			},
+		}
+
+		// Just add conditions, no score or keep the existing score
+		addOnPlacementScore.Status.Conditions = conditions
```

Comment:
Should consider using the `"k8s.io/apimachinery/pkg/api/meta"` library `SetStatusCondition` function to set the condition. For example:

```
meta.SetStatusCondition(&curator.Status.Conditions, metav1.Condition{
				Type:    "HostedClusterCountUpdated",
				Status:  metav1.ConditionFalse,
				Reason:  "HostedClusterCountFailed",
				Message: err.Error(),
```

Reply:
Done. Thank you